draft-barnes-acme.md

```diff
@@ -198,324 +212,362 @@ To revoke a certificate, the client simply sends a revocation request, signed wi
+The following diagram illustrates the relations between resources on an ACME server.  The solid lines indicate link relations, and the dotted lines correspond to relationships expressed in other ways, e.g., the Location header in a 201 (Created) response.
 
-ACME provides three general message types -- "error", "defer", and "statusRequest" -- to cover cases where the server is not able to return a successful result immediately.  If there is a problem that prevents the request from succeeding, then the server sends an error message.  The fields in an error message are as follows:
+~~~~~~~~~~
+             "next"              "next"
+    new-reg ---+----> new-authz ---+----> new-cert    cert-chain
+       .       |          .        |         .            ^
+       .       |          .        |         .            | "up"
+       V       |          V        |         V            |
+      reg* ----+        authz -----+       cert-----------+
+                         . ^
+                         . | "up"
+                         V |
+                       challenge
 
-type (required, string):
-: "error"
+~~~~~~~~~~
 
-error (required, string):
-: A token from the below list indicating what type of error occurred.
 
-message (optional, string):
-: A human-readable string describing the error.
+The remainder of this section provides the details of how these resources are structured and how the ACME protocol makes use of them.
 
-moreInfo (optional, string):
-: A URL of a resource containing additional human-readable documentation about the error, such as advice on how to revise the request or adjust the client configuration to allow the request to succeed, or documentation of CA issuance policies that describe why the request cannot be fulfilled.
+All ACME requests with a non-empty body MUST encapsulate the body in a JWS object, signed using the account key pair.  The server MUST verify the JWS before processing the request.  (For readability, however, the examples below omit this encapsulation.)  Encapsulating request bodies in JWS provides a simple authentication of requests by way of key continuity.
 
-~~~~~~~~~~
+Note that this implies that GET requests are not authenticated.  Servers MUST NOT respond to GET requests for resources that might be considered sensitive.
 
-{
-  "type": "error",
-  "error": "badCSR",
-  "message": "RSA keys must be at least 2048 bits long",
-  "moreInfo": "https://ca.example.com/documentation/csr-requirements"
-}
+The following table illustrates a typical sequence of requests required to establish a new account with the server, prove control of an identifier, issue a certificate, and fetch an updated certificate some time after issuance:
 
-~~~~~~~~~~
+| Action             | Request        | Response     |
+|:-------------------|:---------------|:-------------|
+| Register           | POST new-reg   | 201 -> reg   |
```

Comment:
What does the `->` represent in this chart? I assume it's 'return a body of this resource type,' not 'redirect to this URL?'